### PR TITLE
Improve container routing & dependencies bump

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | yes
| Tickets          | MANAGER-9805, MANAGER-9806
| License          | BSD 3-Clause

## Description

- bump vite to v3
- bump react to v18
- bump react-router-dom to v6
- move vite configuration to external package
- improve container routing
- remove unused fragment/navbar to avoid installing multiple react version on the monorepo

### CODE REVIEW

File: packages/components/ovh-shell/src/client/api.ts [hunk 1]
```diff
@@ -44,8 +44,8 @@ export default function exposeApi(shellClient: ShellClient) {
         }),
     },
     routing: {
-      init: () =>
-        window.addEventListener('hashchange', () => {
+      init: () => {
+        window.addEventListener('hashchange', (e) => {
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/components/ovh-shell/src/client/api.ts]: was using the event to log during dev & forgot about removing it, fixed

File: packages/components/ovh-shell/src/plugin/index.ts [hunk 2]
```diff
@@ -4,4 +4,4 @@ import { i18n } from './i18n';
 import routing from './routing';
```
Comment (mohammed-zahaf) [inline packages/components/ovh-shell/src/plugin/index.ts:4]: i don't see any usage of imported `routing`, should be removed too ?

File: packages/components/ovh-shell/src/message-bus/direct-client.ts [hunk 3]
```diff
@@ -26,4 +26,9 @@ export default class DirectClientMessageBus implements IMessageBus {
   onReceive<T>(callback: (message: T) => void): void {
     this.listeners.push(callback);
   }
+
+  cleanup(): void {
+    this.listeners = [];
```
Comment (mohammed-zahaf) [inline packages/components/ovh-shell/src/message-bus/direct-client.ts:31]: May be we could implement something to unregister listener before set array reference ?

**Review by cbourgois — APPROVED** [review]

**Comment by frenautvh** [comment]
It has been further improved on the feat/core-dev branch. So i'm closing this PR.
A new PR will be re-opened with the improved version in the near future

**Pull request closed without merging**